File: src/test/java/pl/patrykdolata/lifemanager/mapper/AccountMapperTest.java
package pl.patrykdolata.lifemanager.mapper;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import pl.patrykdolata.lifemanager.TestUtils;
import pl.patrykdolata.lifemanager.domain.AccountEntity;
import pl.patrykdolata.lifemanager.model.NewAccount;

import java.math.BigDecimal;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.junit.jupiter.api.Assertions.*;

@SpringBootTest
class AccountMapperTest {

    @Autowired
    private AccountMapper mapper;

    @BeforeEach
    public void init() {
        TestUtils.withProperPrincipal();
    }

    @Test
    void toEntity_whenNullNewAccountModel_expectNull() {
        NewAccount account = null;
        AccountEntity result = mapper.toEntity(account);

        assertNull(result);
    }

    @Test
    void toEntity_whenProperNewAccountModel_expectProperEntity() {
        NewAccount account = new NewAccount("acc", new BigDecimal("12.5"));
        AccountEntity result = mapper.toEntity(account);

        assertNotNull(result);
        assertThat(result.getName(), is(account.getName()));
        assertThat(result.getBalance(), is(account.getStartBalance()));
        assertThat(result.getUser().getId(), is(TestUtils.ID));
    }
}

File: src/test/kotlin/pl/patrykdolata/lifemanager/mapper/AccountMapperTest.kt
package pl.patrykdolata.lifemanager.mapper

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import pl.patrykdolata.lifemanager.TestUtils
import pl.patrykdolata.lifemanager.TestUtils.withProperPrincipal
import pl.patrykdolata.lifemanager.domain.AccountEntity
import pl.patrykdolata.lifemanager.domain.UserEntity
import pl.patrykdolata.lifemanager.model.Account
import java.math.BigDecimal

@SpringBootTest
class AccountMapperTest {

    @Autowired
    private lateinit var mapper: AccountMapper

    @BeforeEach
    fun init() {
        withProperPrincipal()
    }

    @Test
    fun `Mapping null NewAccount should throw NPE`() {
        val account: Account? = null
        assertThrows<NullPointerException> { mapper.toEntity(account!!) }
    }

    @Test
    fun `Mapping proper NewAccount should return proper AccountEntity`() {
        val account = Account("acc", BigDecimal("12.5"))
        val result = mapper.toEntity(account)
        assertNotNull(result)
        assertThat(result.name, equalTo(account.name))
        assertThat(result.balance, equalTo(account.balance))
        assertThat(result.user.id, equalTo(TestUtils.ID))
    }

    @Test
    fun `Mapping null AccountEntity should throw NPE`() {
        val accountEntity: AccountEntity? = null
        assertThrows<NullPointerException> { mapper.toModel(accountEntity!!) }
    }

    @Test
    fun `Mapping proper AccountEntity should return proper Account`() {
        val accountEntity = AccountEntity(UserEntity())
        accountEntity.apply {
            id = 5L
            name = "acc1"
            balance = BigDecimal(12.5)
        }
        val result = mapper.toModel(accountEntity)
        assertNotNull(result)
        assertThat(result.name, equalTo(accountEntity.name))
        assertThat(result.balance, equalTo(accountEntity.balance))
    }
}
